packages/components/pie-toggle-switch/src/index.ts

```diff
@@ -25,38 +29,67 @@ export class PieToggleSwitch extends RtlMixin(LitElement) implements ToggleSwitc
     @property({ type: Boolean, reflect: true })
     public isDisabled = false;
 
+    @property({ type: Object })
+    public label!: LabelProps;
+
     static styles = unsafeCSS(styles);
 
     onToggleChange (event: Event) {
         const target = event?.target as HTMLInputElement;
         this.isChecked = target.checked;
-        this.dispatchEvent(new CustomEvent(EVENT_TOGGLE_SWITCH_CHANGED, { detail: this.isChecked }));
+        this.dispatchEvent(new CustomEvent(ON_EVENT_TOGGLE_SWITCH_CHANGED, { detail: this.isChecked }));
+    }
+
+    /**
+     * Renders the label for a toggle switch if provided.
+     * You can only pass in either a leading or trailing label, if you pass both in
+     * as true nothing renders.
+     *
+     * @private
+     */
+    private renderToggleSwitchLabel (): TemplateResult {
+        const { label } = this;
+
+        if (label?.text && (label?.position === 'leading' || label?.position === 'trailing')) {
```

Comment:
if we go with my suggestion above, this could become

```suggestion
        if (label?.text && (labelPoisitons.include(label.position))) {
```